Align device tracker documentation with core changes (#3118)
home-assistant/developers.home-assistant: docs/core/entity/device-tracker.md

## BaseScannerEntity

A BaseScannerEntity reports the connected state of a device, for example to a bluetooth beacon. If the device is connected the BaseScannerEntity's state will be the name of the associated zone, e.g `home` if associated with the home zone, and if the device is not connected the state will be `not_home`.

> The base class sets the `in_zones` state attribute will be populated with the `entity_id` of the associated zone and zones containing it when the device is connected and empty when not connected.

Reviewer: ```suggestion
The base class sets the `in_zones` state attribute. It will be populated with the `entity_id` of the associated zone and zones containing it when the device is connected and empty when not connected.
```

| Name          | Type                         | Default             | Description                         |
| ------------- | ---------------| ------------------- | ----------------------------------- |
| is_connected  | `bool \| None` | **Required**        | The connection state of the device. |
| source_type   | `SourceType`   | **Required**        | The source type of the device.      |

## ScannerEntity

A ScannerEntity reports the connected state of a device on the local network. If the device is connected the ScannerEntity's state will be the name of the associated zone, e.g `home` if associated with the home zone, and if the device is not connected the state will be `not_home`.

> The base class sets the `in_zones` state attribute will be populated with the `entity_id` of the associated zone and zones containing it when the device is connected and empty when not connected.

Reviewer: ```suggestion
The base class sets the `in_zones` state attribute. It will be populated with the `entity_id` of the associated zone and zones containing it when the device is connected and empty when not connected.
```

## TrackerEntity

> A TrackerEntity tracks the location of a device and reports it either as a location name, a zone name or `home` or `not_home` states. A TrackerEntity normally receives GPS coordinates to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state.

Reviewer: Should we remove the mention of `location name` since we've deprecated that now?

Reviewer: ```suggestion
A TrackerEntity tracks the location of a device and reports it either as a zone name or `home` or `not_home` states. A TrackerEntity normally receives GPS coordinates to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state.
```

Author: Fixed

> A TrackerEntity tracks the location of a device and reports it either as a location name, a zone name or `home` or `not_home` states. A TrackerEntity normally receives GPS coordinates to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state.

Reviewer: I'm not sure we should say "normally receives GPS coordinates". I'd write:

```suggestion
A TrackerEntity tracks the location of a device and reports it either as a location name, a zone name or `home` or `not_home` states. A TrackerEntity can use GPS coordinates or a list of `Zone` entity_ids to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state.
```

Author: Fixed

> A TrackerEntity tracks the location of a device and reports it either as a location name, a zone name or `home` or `not_home` states. A TrackerEntity normally receives GPS coordinates to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state.

Reviewer: We should also mention the priority between `in_zones` and `latitude` + `longitude` here. The `in_zones` attribute will take priority.

Author: Fixed

> A TrackerEntity tracks the location of a device and reports it either as a zone name or `home` or `not_home` states. A TrackerEntity can use GPS coordinates or a list of `Zone` entity_ids to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state. If both `in_zones` and `latitude` + `longitude` are present, `in_zones` has priority. If `in_zones` is not provided, the base class calculates an `in_zones` list, including both active anc passive zone the device is currently in. The state of the entity is the first active zone in the `in_zones` list, `home` if the first active zone is the home zone or `not_home` if there is no active zone in the `in_zones` list.

Author: Fixed

## TrackerEntity

> A TrackerEntity tracks the location of a device and reports it either as a zone name or `home` or `not_home` states. A TrackerEntity can use GPS coordinates or a list of `Zone` entity_ids to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state. If both `in_zones` and `latitude` + `longitude` are present, `in_zones` has priority. If `in_zones` is not provided, the base class calculates an `in_zones` list, including both active anc passive zone the device is currently in. The state of the entity is the first active zone in the `in_zones` list, `home` if the first active zone is the home zone or `not_home` if there is no active zone in the `in_zones` list.

Reviewer: ```suggestion
A TrackerEntity tracks the location of a device and reports it either as a zone name or `home` or `not_home` states. A TrackerEntity can use GPS coordinates or a list of `Zone` entity_ids to determine its state. Either `in_zones` or `latitude` and `longitude` should be set to report state. If both `in_zones` and `latitude` + `longitude` are present, `in_zones` has priority. If `in_zones` is not provided, the base class calculates an `in_zones` list, including both the active and passive zones the device is currently in. The state of the entity is either the first active zone in the `in_zones` list, `home` if the first active zone is the home zone or `not_home` if there is no active zone in the `in_zones` list.
```

Author: Fixed

Properties should always only return information from memory and not do I/O (like network requests). Implement `update()` or `async_update()` to fetch data.
:::

| Name              | Type                | Default          | Description                              |
| ----------------- | --------------------| ---------------- | ---------------------------------------- |
| in_zones          | `list[str] \| None` | `None`           | The zones the device is in, ignored if latitude and longitude are not `None` |
| latitude          | `float \| None`     | `None`           | The latitude coordinate of the device.   |
| location_accuracy | `float`             | `0`              | The location accuracy (m) of the device. |
| longitude         | `float \| None`     | `None`           | The longitude coordinate of the device.  |

> | source_type       | SourceType          | `SourceType.GPS` | The source type of the device.           |

Reviewer 2: While looking at this and at the mobile code, I wonder if there was any reason to not use a polymorphic message there? To have a `type` field that by design exclude in_zones and latitude/longitude from the same message.

Reviewer 2: It is indeed a breaking change.

Reviewer: Do you mean it would have been a breaking change if we'd use a message with a "type" field?

Reviewer 2: Moving away from a LocationUpdate message to something more like 

```
sealed interface UpdateLocation {
    @SerialName("zone_update")
    data class ZoneUpdateLocation(
        /**
         * Deprecated in core 2026.6.0+; use [inZones] instead. Kept for backwards compatibility.
         */
        val locationName: String?,
        /**
         * Full zone entity IDs (e.g. `zone.home`) the device is currently in. Replaces
         * [locationName] in core 2026.6.0+, should be `null` when using core <2026.6.0.
         */
        val inZones: List<String>?,
    ): UpdateLocation

    @SerialName("coordinate_update")
    data class CoordinateUpdateLocation(
        val gps: List<Double>?,
        val gpsAccuracy: Int?,
        val speed: Int?,
        val altitude: Int?,
        val course: Int?,
        val verticalAccuracy: Int?,
    ): UpdateLocation
}
```
Where we have dedicated messages for each specific case. `type` here would have `coordinate_update` or `zone_update`.
